src/NzbDrone.Core/Qualities/QualityModel.cs

```diff
@@ -10,7 +10,36 @@ public class QualityModel : IEmbeddedDocument, IEquatable<QualityModel>
         public Revision Revision { get; set; }
 
         [JsonIgnore]
-        public QualitySource QualitySource { get; set; }
+        public QualityDetectionSource QualityDetectionSource { get; set; }
+
+        [JsonIgnore]
+        public QualitySource QualitySource
+        {
+            get
```

Comment:
Yeah, I've moved it to `AugmentQuality` since it's not used anywhere else anyways.

I think separating source and resolution and breaking things up makes a lot of sense, though definitely a big impact.

> because BRRip = Bluray.480p. BRRip is 'Bluray', but the default resolution isn't guaranteed.

Is a great example where this would shine, as would `Bluray` without a resolution, we default to 720p, but we're just guessing.

Not sure if you agree, but I'd like to tackle that work separately.

Reply:
Agreed, as long as it takes steps in the right direction. You're introducing QualityDetectionSource, which I'd rather not see but I'm ok with it as intermediate, just gotta make sure it doesn't get used elsewhere in the app.